src/result_collection.jl

```diff
@@ -100,7 +103,11 @@ function collect_results!(filename, folder;
         mtimes = Dict{String,Float64}()
     else
         verbose && @info "Loading existing result collection..."
-        data = wload(filename)
+        if isnothing(load_function_wrapper)
+            data = wload(filename)
+        else
+            data = load_function_wrapper(filename)
```

Comment:
Why are we using `nothing` here? Why don't we have the keyword itself be the function? Make `load_function = wload` and then use `load_function` throughout the source code.

Reply:
If you take a look at the pull request I mention why underneath the code snippet. Probably should have included it farther up in the pull request. 

> But I had difficulties getting this to work with to_data_row for the specialized JLD2 case. In that case it doesn't expect wload (as far as I could tell, I'm not an expert on JLD2 though) so that as a default doesn't work. But I am happy to refactor this PR if there is a better or more idiomatic way to achieve this.

Reply:
Sorry, but I don't understand the problem. Please explain it again. The way I see the code right now, if `load_function` is nothing, it becomes `wload`. Otherwise, it becomes a user-provided function. So... `load_function` can be by default `wload`. That's what it becomes anyways. So I am overall confused because in the current PR `wload` is the default but you say it doesn't work as the default.

The quote you quoted: what does "this" mean in its first sentence?

Reply:
> Sorry, but I don't understand the problem. Please explain it again. The way I see the code right now, if `load_function` is nothing, it becomes `wload`. Otherwise, it becomes a user-provided function. So... `load_function` can be by default `wload`. That's what it becomes anyways. So I am overall confused because in the current PR `wload` is the default but you say it doesn't work as the default.
> 
> The quote you quoted: what does "this" mean in its first sentence?

Yeah sorry for the confusion, I probably made this more complicated than it needed to be. I've pushed some new changes based on your advice. Thanks for your patience. Please take a look and let me know what you think.